bin/clover/src/pipelines/azure/funcs/management/discover.ts

```diff
@@ -0,0 +1,273 @@
+  const tenantId = requestStorage.getEnv("AZURE_TENANT_ID");
+  const clientId = requestStorage.getEnv("AZURE_CLIENT_ID");
+  const clientSecret = requestStorage.getEnv("AZURE_CLIENT_SECRET");
+
+  if (!tenantId || !clientId || !clientSecret) {
+    throw new Error("Azure credentials not found");
+  }
+
+  const subscriptionId = _.get(
+    component.properties,
+    ["domain", "subscriptionId"],
+    "",
+  );
+  const resourceGroup = _.get(
+    component.properties,
+    ["domain", "resourceGroup"],
+    "",
+  );
+  const resourceType = _.get(
+    component.properties,
+    ["domain", "extra", "AzureResourceType"],
+    "",
+  );
+  const apiVersion = _.get(
+    component.properties,
+    ["domain", "extra", "apiVersion"],
+    "2023-01-01",
+  );
+  const propUsageMapJson = _.get(
+    component.properties,
+    ["domain", "extra", "PropUsageMap"],
+    "{}",
+  );
+
+  if (!subscriptionId) {
+    return {
+      status: "error",
+      message: "subscriptionId is required in domain",
+    };
+  }
+
+  if (!resourceType) {
+    return {
+      status: "error",
+      message: "AzureResourceType not found in domain.extra",
+    };
+  }
+
+  // Convert Azure::Service::Resource to Microsoft.Service/resources format
+  const parts = resourceType.split("::");
+  if (parts.length !== 3 || parts[0] !== "Azure") {
+    return {
+      status: "error",
+      message: `Invalid Azure resource type format: ${resourceType}`,
+    };
+  }
+
+  const providerNamespace = `Microsoft.${parts[1]}`;
+  const resourceTypeName = parts[2];
+
+  // Parse PropUsageMap to know which properties are writable
+  let writableProperties: Set<string>;
+  let createOnlyProperties: Set<string>;
+  try {
+    const propUsageMap = JSON.parse(propUsageMapJson);
+    writableProperties = new Set([
+      ...(propUsageMap.createOnly || []),
+      ...(propUsageMap.updatable || []),
+    ]);
+    createOnlyProperties = new Set(propUsageMap.createOnly || []);
+  } catch (e) {
+    console.warn(
+      `Failed to parse PropUsageMap for ${resourceType}, using empty set:`,
+      e,
+    );
+    writableProperties = new Set();
+    createOnlyProperties = new Set();
+  }
+
+  console.log(`Discovering ${resourceType} resources...`);
+
+  const token = await getAzureToken(tenantId, clientId, clientSecret);
+
+  // Build refinement filter from domain properties
+  const refinement = _.cloneDeep(thisComponent.properties.domain);
+  delete refinement["extra"];
+  // Remove any empty values, as they are never refinements
+  for (const [key, value] of Object.entries(refinement)) {
+    if (_.isEmpty(value)) {
+      delete refinement[key];
+    } else if (_.isPlainObject(value)) {
+      refinement[key] = _.pickBy(
+        value,
+        (v) => !_.isEmpty(v) || _.isNumber(v) || _.isBoolean(v),
+      );
+      if (_.isEmpty(refinement[key])) {
+        delete refinement[key];
+      }
+    }
+  }
+
+  // Build the list URL based on whether resourceGroup is specified
```

Comment:
Maybe it would make more sense to just store this on the asset in the extra prop?